File: java/app/src/main/java/com/huawei/codelabpush/MyPushService.java

package com.huawei.codelabpush;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.huawei.hms.push.HmsMessageService;
import com.huawei.hms.push.RemoteMessage;

public class MyPushService extends HmsMessageService {
    private static final String TAG = "PushDemoLog";
    @Override
    public void onNewToken(String token, Bundle bundle) {
        super.onNewToken(token);
        Log.i(TAG, "receive token:" + token);
        sendTokenToDisplay(token);
    }
    @Override
    public void onTokenError(Exception e) {
        super.onTokenError(e);
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        if (remoteMessage.getData().length() > 0) {
            Log.i(TAG, "Message data payload: " + remoteMessage.getData());
        }
        if (remoteMessage.getNotification() != null) {
            Log.i(TAG, "Message Notification Body: " + remoteMessage.getNotification().getBody());
        }
    }

    @Override
    public void onMessageSent(String s) {
    }

    @Override
    public void onSendError(String s, Exception e) {
    }

    private void sendTokenToDisplay(String token) {
        Intent intent=new Intent("com.huawei.codelabpush.ON_NEW_TOKEN");
        intent.putExtra("token", token);
        sendBroadcast(intent);
    }
}

File: kotlin/app/src/main/java/com/huawei/loveandshare/MyPushService.kt
package com.huawei.loveandshare

import android.content.Intent
import android.util.Log
import com.huawei.hms.push.HmsMessageService
import com.huawei.hms.push.RemoteMessage

class MyPushService : HmsMessageService() {
    override fun onNewToken(token: String?) {
        Log.i(TAG, "receive token:$token")
        if (!token.isNullOrEmpty()) {
            sendTokenToDisplay(token)
        }
    }

    override fun onTokenError(e: Exception) {
        super.onTokenError(e)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {
        if (remoteMessage == null) {
            return
        }
        if (remoteMessage.data?.length ?: -1 > 0) {
            Log.i(TAG, "Message data payload: " + remoteMessage.data)
        }
        if (remoteMessage.notification != null) {
            Log.i(TAG, "Message Notification Body: " + remoteMessage.notification.body)
        }
    }

    override fun onMessageSent(msgId: String?) {}

    override fun onSendError(msgId: String?, exception: Exception?) {}

    private fun sendTokenToDisplay(token: String) {
        val intent = Intent("com.huawei.codelabpush.ON_NEW_TOKEN")
        intent.putExtra("token", token)
        sendBroadcast(intent)
    }

    companion object {
        private val TAG: String? = "PushDemoLog"
    }
}